### ISSUE TITLE
Improvements to the Spline

### ISSUE THREAD

**Issue by PhyresiCompany** [FlaxEngine/FlaxEngine#1125]
**SplineModel Improvements**
- In Unity Spline you can add as many meshes as you want and you can set a percentage for them.
![Captura de Tela (135)](https://github.com/FlaxEngine/FlaxEngine/assets/101017436/8df86449-b875-429a-b590-56fc5ebc3afb)

- Through Distance you can define an exact or random distance. It is very useful to create posts with a distance between them.
![Captura de Tela (136)](https://github.com/FlaxEngine/FlaxEngine/assets/101017436/9ab6817b-356e-46d4-b602-d2bd5c1fc63b)

- You can randomize any Transform value (Position/Rotation/Scale) or you can use an exact value for the mesh
![Captura de Tela (137)](https://github.com/FlaxEngine/FlaxEngine/assets/101017436/c93b97db-fcba-4ab5-b9c0-df43ee2a609a)

**Usability improvement for curve editing**

- In Unity, to shape the spline you only use the mouse and it has a separate gizmo

https://github.com/FlaxEngine/FlaxEngine/assets/101017436/536070ed-5522-4389-b9a4-65a22292e42f

- And you can create on top of other surfaces/objects

https://github.com/FlaxEngine/FlaxEngine/assets/101017436/e99b3122-8aaa-4e93-901c-0407702d1495

- In Flax you have to click on the spline keyframe and use the shortcut CTRL+D to clone the keyframe and then move it to the desired position

https://github.com/FlaxEngine/FlaxEngine/assets/101017436/a0d1c96a-7f52-4ee8-a942-160e48d5cafc

**General improvements**

- Create a new spline actor that allows adding actors to it in addition to a Model to give more freedom in using the Spline

- Allow the splines to connect with each other to create this effect
![depositphotos_493176784-stock-photo-top-down-view-of-road](https://github.com/FlaxEngine/FlaxEngine/assets/101017436/7f523178-675b-419a-9706-da9c7ce32ae3)

- Lets you change the mesh for just a specific spline keyframe. This is very useful for T intersections / Crossings in traffic
![post-regras-cruzamento-preferencia](https://github.com/FlaxEngine/FlaxEngine/assets/101017436/b0c4970f-0b66-4685-95de-7856ca29b600)

**Comment by RuanLucasGD** [comment]
Yes, i wold also like to have some of these features on Spline system.

****

**Spline:**
* Add simple form to add points, (only clicking in surfaces?)
* Add option to change spline mode using dropdown (smooth / linear)

**Spline Model**
* Have Distance between meshs
* Add a option to enable/disable meshs deformation
* Add hability to change mesh os specific part of Spline Model.
* Add option to set meshs on multiple intersections intersections (use case: roads)

**Additional Actor -> Spline Terrain**

* Gets and change terrain height to create roads (Storage the height on custom texture from each Spline Terrain to not change original terrain?)

**Comment by RuanLucasGD** [comment]
#1292
